src/download_and_link.rs

```diff
@@ -65,6 +65,31 @@ impl DownloadAndLinkHooks for () {
 	type Error = Infallible;
 }
 
+/// Options for which dependencies to install.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default)]
+pub enum InstallDependenciesMode {
+	/// Install all dependencies.
+	#[default]
+	All,
+	/// Install only dependencies, not dev_dependencies.
+	Prod,
+	/// Install only dev_dependencies.
+	Dev,
+}
+impl PartialEq<DependencyType> for InstallDependenciesMode {
+	fn eq(&self, other: &DependencyType) -> bool {
```

Comment:
```suggestion

impl InstallDependenciesMode {
	fn fits(&self, dep_ty: &DependencyType) -> bool {
```

Let's not use PartialEq for this.